### PR TITLE
Automatic port of System.Reflection documentation

### PR DESCRIPTION
**Pull request by carlossanlop**
Ported with [DocsPortingTool](https://github.com/carlossanlop/DocsPortingTool).

### CODE REVIEW

File: xml/System.Reflection.Metadata.Ecma335/MethodBodyStreamEncoder.xml [hunk 1]
```diff
@@ -170,6 +180,8 @@
         <summary>Encodes a method body and adds it to the method body stream, using the provided code size, maximum stack size, number of exception regions, local variables' signature handle, method body attributes and allowing to indicate whether the exception regions should be encoded in small format or not.</summary>
         <returns>The offset of the encoded body within the method body stream.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="codeSize" />, <paramref name="exceptionRegionCount" />, or <paramref name="maxStack" /> is out of allowed range.</exception>
```
Comment (gewarren) [inline xml/System.Reflection.Metadata.Ecma335/MethodBodyStreamEncoder.xml:184]:
```suggestion
          <paramref name="codeSize" />, <paramref name="exceptionRegionCount" />, or <paramref name="maxStack" /> is out of the allowed range.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/MethodBodyStreamEncoder.xml [hunk 2]
```diff
@@ -211,6 +223,8 @@
         <summary>Encodes a method body and adds it to the method body stream, using the provided code size, maximum stack size, number of exception regions, local variables' signature handle, method body attributes, allowing to indicate whether the exception regions should be encoded in small format or not, and allowing to indicate whether the method should allocate from the dynamic local memory pool or not.</summary>
         <returns>The offset of the encoded body within the method body stream.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="codeSize" />, <paramref name="exceptionRegionCount" />, or <paramref name="maxStack" /> is out of allowed range.</exception>
```
Comment (gewarren) [inline xml/System.Reflection.Metadata.Ecma335/MethodBodyStreamEncoder.xml:227]:
```suggestion
          <paramref name="codeSize" />, <paramref name="exceptionRegionCount" />, or <paramref name="maxStack" /> is out of the allowed range.</exception>
```

File: xml/System.Reflection/MethodInfo.xml [hunk 3]
```diff
@@ -324,7 +324,7 @@ For a list of the invariant conditions for other terms used in generic reflectio
       <Parameters />
       <Docs>
         <typeparam name="T">To be added.</typeparam>
-        <summary>To be added.</summary>
+        <summary>Creates a delegate of the given type 'T' from this method.</summary>
```
Comment (gewarren) [inline xml/System.Reflection/MethodInfo.xml:327]:
```suggestion
        <summary>Creates a delegate of type <typeparamref name="T" /> from this method.</summary>
```

File: xml/System.Reflection/MethodInfo.xml [hunk 4]
```diff
@@ -366,7 +366,7 @@ For a list of the invariant conditions for other terms used in generic reflectio
       <Docs>
         <typeparam name="T">To be added.</typeparam>
         <param name="target">To be added.</param>
-        <summary>To be added.</summary>
+        <summary>Creates a delegate of the given type 'T' with the specified target from this method.</summary>
```
Comment (gewarren) [inline xml/System.Reflection/MethodInfo.xml:369]:
```suggestion
        <summary>Creates a delegate of type <typeparamref name="T" /> with the specified target from this method.</summary>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 5]
```diff
@@ -83,6 +84,7 @@
         <summary>Calculates a HasConstant coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:87]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 6]
```diff
@@ -114,6 +116,7 @@
         <summary>Calculates a HasCustomAttribute coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:119]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 7]
```diff
@@ -145,6 +148,7 @@
         <summary>Calculates a HasCustomDebugInformation coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:151]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 8]
```diff
@@ -176,6 +180,7 @@
         <summary>Calculates a HasDeclSecurity coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:183]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 9]
```diff
@@ -207,6 +212,7 @@
         <summary>Calculates a HasFieldMarshal coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:215]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 10]
```diff
@@ -238,6 +244,7 @@
         <summary>Calculates a HasSemantics coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:247]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 11]
```diff
@@ -269,6 +276,7 @@
         <summary>Calculates an implementation coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:279]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 12]
```diff
@@ -300,6 +308,7 @@
         <summary>Calculates a MemberForwarded coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:311]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 13]
```diff
@@ -331,6 +340,7 @@
         <summary>Calculates a MemberRefParent coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:343]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 14]
```diff
@@ -362,6 +372,7 @@
         <summary>Calculates a MethodDefOrRef coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:375]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 15]
```diff
@@ -393,6 +404,7 @@
         <summary>Calculates a ResolutionScope coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:407]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 16]
```diff
@@ -424,6 +436,7 @@
         <summary>Calculates a TypeDefOrRef coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:439]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 17]
```diff
@@ -455,6 +468,7 @@
         <summary>Calculates a TypeDefOrRefOrSpec coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:471]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml [hunk 18]
```diff
@@ -486,6 +500,7 @@
         <summary>Calculates a TypeOrMethodDef coded index for the specified handle.</summary>
         <returns>To be added.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">Unexpected handle kind.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/CodedIndex.xml:503]:
```suggestion
        <exception cref="T:System.ArgumentException">The handle type is unexpected.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/InstructionEncoder.xml [hunk 19]
```diff
@@ -87,6 +87,14 @@
         <param name="label">Label of the target location in instruction stream.</param>
         <summary>Encodes a branch instruction.</summary>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentException">
+          <paramref name="code" /> is not a branch instruction.</exception>
+        <exception cref="T:System.ArgumentException">
+          <paramref name="label" /> was not defined by this encoder.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/InstructionEncoder.xml:93]:
```suggestion
        <exception cref="T:System.ArgumentException">
          <paramref name="code" /> is not a branch instruction.

-or-
                    
<paramref name="label" /> was not defined by this encoder.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/MetadataBuilder.xml [hunk 20]
```diff
@@ -2179,6 +2250,10 @@ Use this method to reduce allocations if the approximate number of bytes is know
 
           ]]></format>
         </remarks>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="heap" /> is not a valid heap index.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="byteCount" /> is negative.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/MetadataBuilder.xml:2256]:
```suggestion
        <exception cref="T:System.ArgumentOutOfRangeException">
          <paramref name="heap" /> is not a valid heap index.

-or-

<paramref name="byteCount" /> is negative.</exception>
```

File: xml/System.Reflection.Metadata.Ecma335/MetadataBuilder.xml [hunk 21]
```diff
@@ -2179,6 +2250,10 @@ Use this method to reduce allocations if the approximate number of bytes is know
 
           ]]></format>
         </remarks>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="heap" /> is not a valid heap index.</exception>
```
[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System.Reflection.Metadata.Ecma335/MetadataBuilder.xml:2254]: I may be able to add this feature to the tool.

File: xml/System.Reflection.Metadata.Ecma335/MetadataBuilder.xml [hunk 22]
```diff
@@ -2218,6 +2293,10 @@ Use this method to reduce allocations if the approximate number of rows is known
 
           ]]></format>
         </remarks>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="table" /> is not a valid table index.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="rowCount" /> is negative.</exception>
```
Comment (carlossanlop) [inline xml/System.Reflection.Metadata.Ecma335/MetadataBuilder.xml:2299]:
```suggestion
        <exception cref="T:System.ArgumentOutOfRangeException">
          <paramref name="table" /> is not a valid table index.

-or-

<paramref name="rowCount" /> is negative.</exception>
```

**Comment by carlossanlop** [comment]
@gewarren thanks so much for the review. Would you mind taking a look again?

**Comment by carlossanlop** [comment]
@dotnet/docs since @gewarren is away. Can I please get a sign-off?

**Pull request merged**